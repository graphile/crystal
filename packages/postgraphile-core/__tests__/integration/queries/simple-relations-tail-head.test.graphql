query{
  allCompoundKeysList {
    personId1
    personId2
    extra
    personByPersonId1 {
      name
      email
    }
    personByPersonId2 {
      name
      email
    }
  }
  allForeignKeysList {
    personId
    compoundKey1
    compoundKey2
    personByPersonId {
      name
      email
    }
    compoundKeyByCompoundKey1AndCompoundKey2 {
      personId1
      personId2
      extra
    }
  }
}
