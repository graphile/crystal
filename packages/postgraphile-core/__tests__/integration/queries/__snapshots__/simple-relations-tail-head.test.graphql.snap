// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`execute 1`] = `
Object {
  "data": Object {
    "allCompoundKeysList": Array [
      Object {
        "extra": null,
        "personByPersonId1": Object {
          "email": "john.smith@email.com",
          "name": "John Smith",
        },
        "personByPersonId2": Object {
          "email": "sara.smith@email.com",
          "name": "Sara Smith",
        },
        "personId1": 1,
        "personId2": 2,
      },
      Object {
        "extra": false,
        "personByPersonId1": Object {
          "email": "sara.smith@email.com",
          "name": "Sara Smith",
        },
        "personByPersonId2": Object {
          "email": "john.smith@email.com",
          "name": "John Smith",
        },
        "personId1": 2,
        "personId2": 1,
      },
      Object {
        "extra": null,
        "personByPersonId1": Object {
          "email": "sara.smith@email.com",
          "name": "Sara Smith",
        },
        "personByPersonId2": Object {
          "email": "budd.deey@email.com",
          "name": "Budd Deey",
        },
        "personId1": 2,
        "personId2": 3,
      },
      Object {
        "extra": true,
        "personByPersonId1": Object {
          "email": "sara.smith@email.com",
          "name": "Sara Smith",
        },
        "personByPersonId2": Object {
          "email": "joe.tucker@email.com",
          "name": "Joe Tucker",
        },
        "personId1": 2,
        "personId2": 5,
      },
      Object {
        "extra": true,
        "personByPersonId1": Object {
          "email": "kathryn.ramirez@email.com",
          "name": "Kathryn Ramirez",
        },
        "personByPersonId2": Object {
          "email": "budd.deey@email.com",
          "name": "Budd Deey",
        },
        "personId1": 4,
        "personId2": 3,
      },
      Object {
        "extra": false,
        "personByPersonId1": Object {
          "email": "kathryn.ramirez@email.com",
          "name": "Kathryn Ramirez",
        },
        "personByPersonId2": Object {
          "email": "kathryn.ramirez@email.com",
          "name": "Kathryn Ramirez",
        },
        "personId1": 4,
        "personId2": 4,
      },
    ],
    "allForeignKeysList": Array [
      Object {
        "compoundKey1": 2,
        "compoundKey2": 1,
        "compoundKeyByCompoundKey1AndCompoundKey2": Object {
          "extra": false,
          "personId1": 2,
          "personId2": 1,
        },
        "personByPersonId": Object {
          "email": "joe.tucker@email.com",
          "name": "Joe Tucker",
        },
        "personId": 5,
      },
      Object {
        "compoundKey1": 4,
        "compoundKey2": 4,
        "compoundKeyByCompoundKey1AndCompoundKey2": Object {
          "extra": false,
          "personId1": 4,
          "personId2": 4,
        },
        "personByPersonId": null,
        "personId": null,
      },
      Object {
        "compoundKey1": 2,
        "compoundKey2": 1,
        "compoundKeyByCompoundKey1AndCompoundKey2": Object {
          "extra": false,
          "personId1": 2,
          "personId2": 1,
        },
        "personByPersonId": null,
        "personId": null,
      },
    ],
  },
}
`;
